### ISSUE TITLE
"Open in Designer" fails when executed from palette

### ISSUE THREAD

**Issue by Zeroji** [andisfar/LaunchQtCreator#24]
Hi, I've added a keyboard shortcut to `launchqtcreator.openinqtdesigner` with the intention to use it while editing a .ui file, however launching the command via a keyboard shortcut or the command palette fails with the message "Cannot read properties of undefined (reading 'fsPath')", with the following trace:

```
2024-11-14 09:38:16.642 [error] TypeError: Cannot read properties of undefined (reading 'fsPath')
	at commandInHandler (c:\Users\Redacted\.vscode\extensions\oralshanecurtis-6510.launchqtcreator-0.1.16\out\extension.js:132:54)
	at T0.h (file:///c:/Users/Redacted/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:32845)
	at T0.$executeContributedCommand (file:///c:/Users/Redacted/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:33695)
	at Zb.S (file:///c:/Users/Redacted/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:31:113896)
	at Zb.Q (file:///c:/Users/Redacted/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:31:113676)
	at Zb.M (file:///c:/Users/Redacted/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:31:112765)
	at Zb.L (file:///c:/Users/Redacted/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:31:111870)
	at gh.value (file:///c:/Users/Redacted/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:31:110667)
	at T.B (file:///c:/Users/Redacted/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:29:732)
	at T.fire (file:///c:/Users/Redacted/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:29:950)
	at no.fire (file:///c:/Users/Redacted/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:31:9399)
	at gh.value (file:///c:/Users/Redacted/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:174:13273)
	at T.B (file:///c:/Users/Redacted/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:29:732)
	at T.fire (file:///c:/Users/Redacted/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:29:950)
	at no.fire (file:///c:/Users/Redacted/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:31:9399)
	at MessagePortMain.<anonymous> (file:///c:/Users/Redacted/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/workbench/api/node/extensionHostProcess.js:174:11562)
	at MessagePortMain.emit (node:events:519:28)
	at MessagePortMain._internalPort.emit (node:electron/js2c/utility_init:2:2619) launchqtcreator.openinqtdesigner {"value":"OralShaneCurtis-6510.launchqtcreator","_lower":"oralshanecurtis-6510.launchqtcreator"}
```

Right-clicking in the editor and then using the "Open in Designer" context menu option works, but I would like the command to work using the currently focused editor so I can use it via a keyboard shortcut.

Here are my system details:
```
Version: 1.94.2 (user setup)
Commit: 384ff7382de624fb94dbaf6da11977bba1ecd427
Date: 2024-10-09T16:08:44.566Z
Electron: 30.5.1
ElectronBuildId: 10262041
Chromium: 124.0.6367.243
Node.js: 20.16.0
V8: 12.4.254.20-electron.0
OS: Windows_NT x64 10.0.22631
```
I'm using the latest version of the extension (v0.1.16).

**Comment by andisfar** [comment]
Thank you for your details. I very much appreciate your reaching out.  I will be fixing this issue in the near future

**Comment by andisfar** [comment]
This is my environment:
Version: 1.96.2 (user setup)
Commit: fabdb6a30b49f79a7aba0f2ad9df9b399473380f
Date: 2024-12-19T10:22:47.216Z
Electron: 32.2.6
ElectronBuildId: 10629634
Chromium: 128.0.6613.186
Node.js: 20.18.1
V8: 12.8.374.38-electron.0
OS: Windows_NT x64 10.0.22631
I found the cause of the command palette failures and have added a check to ensure that the command pallet has a valid file name.  As long as the active editor window is a file that the extension allows to be opened in QtCreator or Designer the function will work from keyboard shortcut, command palette, explorer context menu, or editor context menu.
I will be posting this fix in version 0.1.18, shortly.

**Comment by andisfar** [comment]
The 0.1.18 version which adresses the issue here, has been published to the vscode marketplace and will be available shortly.
